### PR TITLE
benchmark: watch-get for testing unsynced watcher/get contention

### PR DESCRIPTION
**Pull request by heyitsanthony**

### CODE REVIEW

**Comment by heyitsanthony** [comment]
Reproduces behavior reported in #4567

**Comment by gyuho** [comment]
LGTM. Defer to @xiang90

File: tools/benchmark/cmd/watch_get.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+			watchRev = resp.Header.Revision
+		}
+	}
+
+	streams := make([]v3.Watcher, watchGetTotalStreams)
+	for i := range streams {
+		streams[i] = v3.NewWatcher(clients[i%len(clients)])
+	}
+
+	// results from trying to do puts with concurrent watchers
```
Comment (xiang90) [inline tools/benchmark/cmd/watch_get.go]: do serialized get instead of do puts?

**Comment by heyitsanthony** [comment]
all fixed /cc @xiang90

**Comment by xiang90** [comment]
can you show the benchmark results for 10, 10000 unsync watchers?

**Comment by heyitsanthony** [comment]
```
$ tools/benchmark/benchmark watch-get
 80000 / 80000 Booooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo! 100.00%0
Get during watch summary:

Summary:
  Total:        0.9975 secs.
  Slowest:      0.3670 secs.
  Fastest:      0.0004 secs.
  Average:      0.1235 secs.
  Stddev:       0.1722 secs.
  Requests/sec: 3.0075

Response time histogram:
  0.000 [1]     |ââââââââââââââââââââââââââââââââââââââââ
  0.037 [1]     |ââââââââââââââââââââââââââââââââââââââââ
  0.074 [0]     |
  0.110 [0]     |
  0.147 [0]     |
  0.184 [0]     |
  0.220 [0]     |
  0.257 [0]     |
  0.294 [0]     |
  0.330 [0]     |
  0.367 [1]     |ââââââââââââââââââââââââââââââââââââââââ

Latency distribution:
  10% in 0.0030 secs.
  25% in 0.3670 secs.
```

**Comment by heyitsanthony** [comment]
10 watchers
```
tools/benchmark/benchmark watch-get --watchers=10
 80 / 80 Boooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo! 100.00%0
Get during watch summary:

Summary:
  Total:        0.0327 secs.
  Slowest:      0.0003 secs.
  Fastest:      0.0001 secs.
  Average:      0.0001 secs.
  Stddev:       0.0000 secs.
  Requests/sec: 7764.1968

Response time histogram:
  0.000 [1]     |
  0.000 [93]    |âââââââââââââââââââââââââââââ
  0.000 [127]   |ââââââââââââââââââââââââââââââââââââââââ
  0.000 [20]    |ââââââ
  0.000 [4]     |â
  0.000 [5]     |â
  0.000 [1]     |
  0.000 [1]     |
  0.000 [0]     |
  0.000 [1]     |
  0.000 [1]     |

Latency distribution:
  10% in 0.0001 secs.
  25% in 0.0001 secs.
  50% in 0.0001 secs.
  75% in 0.0001 secs.
  90% in 0.0001 secs.
  95% in 0.0002 secs.
  99% in 0.0002 secs.
```

**Comment by xiang90** [comment]
@heyitsanthony OK... I am convinced. Good... LGTM

**Pull request merged**